server/src/main/java/org/elasticsearch/upgrades/SystemIndexMigrator.java

```diff
@@ -534,6 +496,36 @@ private void createIndex(SystemIndexMigrationInfo migrationInfo, ActionListener<
         );
     }
 
+    private void createIndexRetryOnFailure(SystemIndexMigrationInfo migrationInfo, ActionListener<ShardsAcknowledgedResponse> listener) {
+        createIndex(migrationInfo, listener.delegateResponse((l, e) -> {
+            logger.warn("createIndex failed, retrying after removing index [{}] from previous attempt", migrationInfo.getNextIndexName());
+            deleteIndex(migrationInfo, ActionListener.wrap(cleanupResponse -> createIndex(migrationInfo, l.delegateResponse((l3, e3) -> {
```

Comment:
Here, we try to delete the index and retry if there is any error with the prior create call.  I could instead only catch the specific "resource already exists" exception.  However,  I'd rather be more broad here in case there are other valid exceptions that get thrown, where deleting the existing resource is the right thing to do.  Otherwise, we may end up in a non-recoverable state as a result of some scenario we haven't predicted.